src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -226,6 +226,14 @@ protected void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArguments<At
                 return;
             }
 
+            if (arguments.Attribute.IsTargetAttribute(this, AttributeDescription.ExperimentalAttribute)
+                && !SyntaxFacts.IsValidIdentifier((string?)arguments.Attribute.CommonConstructorArguments[0].ValueInternal))
+            {
+                var attrArgumentLocation = arguments.Attribute.GetAttributeArgumentSyntaxLocation(parameterIndex: 0, arguments.AttributeSyntaxOpt);
+                arguments.Diagnostics.DiagnosticBag.Add(ErrorCode.ERR_InvalidExperimentalDiagID, attrArgumentLocation);
+                return;
```

Comment:
>return [](http://example.com/codeflow?start=16&length=6)

Shouldn't we still call ```DecodeWellKnownAttributeImpl``` below?

Reply:
I don't think so. The decoding of experimental attribute is done in `EarlyDecodeWellKnownAttributes`, before we get to `DecodeWellKnownAttribute`

Reply:
> I don't think so. The decoding of experimental attribute is done in EarlyDecodeWellKnownAttributes, before we get to DecodeWellKnownAttribute

First, the code should be robust to future changes and it feels reasonable to me that the fact you are referring to, if it is a fact, won't hold in the future.

Second, I actually see decoding done in ```DecodeWellKnownAttributeImpl``` for a SourceModuleSymbol today,  for example.

Reply:
Yes, you're right. The `return` broke things.